src/reporting/Market.sol

```diff
@@ -444,24 +461,29 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
             return ReportingState.FORKING;
         }
 
+        if (block.timestamp > reportingWindow.getEndTime()) {
+            return ReportingState.AWAITING_FINALIZATION;
+        }
+
         // If a limited dispute bond has been posted we are in some phase of all reporting depending on time
         if (_limitedReportDisputed) {
-            if (block.timestamp > reportingWindow.getEndTime()) {
-                return ReportingState.AWAITING_FINALIZATION;
-            }
             if (reportingWindow.isDisputeActive()) {
-                return ReportingState.ALL_DISPUTE;
+                if (reportingWindow.hasReceivedAllReport(this)) {
+                    return ReportingState.ALL_DISPUTE;
+                } else {
+                    return ReportingState.AWAITING_FINALIZATION;
```

Comment:
Shouldn't this be, `ReportingState.AWAITING_ALL_REPORTING` (or similar)?  Perhaps we are overloading `AWAITING_FINALIZATION` more than I realized and it really is just a generic `PENDING` state.

Reply:
In this case we're in the DISPUTE phase but since no reporting was received we are awaiting finalization so that we will migrate to a new window

Reply:
Isn't this code path in the "all we know is that a limited dispute exists, but we still may be in limited dispute phase"?

Reply:
We know for certain at this point we're in the `ALL DISPUTE` phase and that we've received no reporting during the `ALL REPORT` phase.